h/groups/search.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+
+class GroupAuthFilter(object):
+    def __init__(self, request):
+        self.auth_domain = request.auth_domain
+        self.authenticated_user = request.authenticated_user
+        self.session = request.db
+        self.group_service = request.find_service(name='group')
+
+    def __call__(self, _):
+        groups = set(['__world__'])
+
+        if self.authenticated_user:
+            for group in self.authenticated_user.groups:
+                groups.add(group.pubid)
+
+        for pubid in self.group_service.world_readable_pubids():
```

Comment:
I think it should probably be something like `Group.members.any(User.id == user.id)`, actually.